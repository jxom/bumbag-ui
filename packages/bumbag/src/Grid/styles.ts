import { css, cssClass } from '../styled';
import { theme } from '../utils';

export const Grid = (styleProps) => cssClass`
  display: grid;

  ${
    styleProps.autoFlow &&
    css`
      grid-auto-flow: ${styleProps.autoFlow};
    `
  }
  ${
    styleProps.gap &&
    css`
      grid-gap: ${styleProps.gap};
    `
  }
  ${
    styleProps.template &&
    css`
      grid-template: ${styleProps.template};
    `
  }
  ${
    styleProps.templateAreas &&
    css`
      grid-template-areas: ${styleProps.templateAreas};
    `
  }
  ${
    styleProps.templateColumns &&
    css`
      grid-template-columns: ${styleProps.templateColumns};
    `
  }
  ${
    styleProps.templateRows &&
    css`
      grid-template-rows: ${styleProps.templateRows};
    `
  }
  ${
    styleProps.autoColumns &&
    css`
      grid-auto-columns: ${styleProps.autoColumns};
    `
  }
  ${
    styleProps.autoRows &&
    css`
      grid-auto-rows: ${styleProps.autoRows};
    `
  }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;

export const GridItem = (styleProps) => cssClass`
  display: grid-item;

  ${
    styleProps.area &&
    css`
      grid-area: ${styleProps.area};
    `
  }
  ${
    styleProps.column &&
    css`
      grid-column: ${styleProps.column};
    `
  }
  ${
    styleProps.row &&
    css`
      grid-row: ${styleProps.row};
    `
  }
  ${
    styleProps.columnStart &&
    css`
      grid-column-start: ${styleProps.columnStart};
    `
  }
  ${
    styleProps.columnEnd &&
    css`
      grid-column-end: ${styleProps.columnEnd};
    `
  }
  ${
    styleProps.rowStart &&
    css`
      grid-row-start: ${styleProps.rowStart};
    `
  }
  ${
    styleProps.rowEnd &&
    css`
      grid-row-end: ${styleProps.rowEnd};
    `
  }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
