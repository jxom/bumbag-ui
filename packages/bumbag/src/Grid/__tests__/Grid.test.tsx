import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Grid } from '../Grid';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Grid>Hello world</Grid>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <Grid autoFlow="row" gap="3vw">
        Hello world
      </Grid>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Grid use="p">Hello world</Grid>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Grid props', () => {
      const { result } = renderHook(() => Grid.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Grid>{(GridProps) => <div {...GridProps}>Hello world</div>}</Grid>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Grid.base should render correctly', () => {
    const { container } = render(<Grid>hello world</Grid>, {
      theme: { Grid: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Grid>hello world</Grid>, {
      theme: { Grid: { defaultProps: { autoFlow: 'row', className: 'test' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
