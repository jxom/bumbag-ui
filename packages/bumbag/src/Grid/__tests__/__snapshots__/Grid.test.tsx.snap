// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-0 {
  display: grid;
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<p
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Grid"
>
  Hello world
</p>
`;

exports[`composition hook should return with Grid props 1`] = `
Object {
  "className": "css-x2olgu css-63it0p css-jwhr8m bb-Box bb-Grid",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-0 {
  display: grid;
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Grid"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-0 {
  display: grid;
  grid-auto-flow: row;
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 test bb-Box bb-Grid"
>
  hello world
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-0 {
  display: grid;
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Grid"
>
  Hello world
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-0 {
  display: grid;
  grid-auto-flow: row;
  grid-gap: 3vw;
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  gap: 3vw!important;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Grid"
>
  Hello world
</div>
`;

exports[`theming Grid.base should render correctly 1`] = `
.emotion-0 {
  display: grid;
}

.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Grid"
>
  hello world
</div>
`;
